fbrp/src/fbrp/cmd/up.py

```diff
@@ -43,6 +46,44 @@ def get_proc_names(proc_names, include_deps):
     return proc_names
 
 
+def down_existing(args: argparse.Namespace, names: typing.List[str]):
+    def find_active_proc(system_state):
+        return [
+            name
+            for name in names
+            if name in system_state.procs
+            and system_state.procs[name].state != life_cycle.State.STOPPED
+        ]
+
+    active_proc = find_active_proc(life_cycle.system_state())
+    if not active_proc:
+        return
+
+    if not args.force:
+        util.fail(f"Conflicting processes already running: {', '.join(active_proc)}")
+
+    for name in active_proc:
+        life_cycle.set_ask(name, life_cycle.Ask.DOWN)
+
+    ns = types.SimpleNamespace()
+    ns.cv = threading.Condition()
+    ns.sat = False
+
+    def callback(system_state):
+        if not find_active_proc(system_state):
+            with ns.cv:
+                ns.sat = True
+                ns.cv.notify()
+
+    watcher = life_cycle.system_state_watcher(callback)
+
+    with ns.cv:
+        success = ns.cv.wait_for(lambda: ns.sat, timeout=3.0)
+
+    if not success:
+        util.fail(f"Existing processes did not down in a timely manner.")
+
+
```

Comment:
This is more of a normative comment and can be ignored :) now that i am writing unit tests, i am noticing that we could refactor some of these to make unit testing better. Just something to think about :)

Reply:
I'll keep this in mind :)